packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -146,6 +146,57 @@ exports.sourceNodes = async (
   newState[createSyncToken()] = nextSyncToken
   setPluginStatus(newState)
 
+  // Guarantee at least one asset exists.
+  // The subsequent GraphQL fragments added by fragmeent.js have an infered model to build against.
+  // With no assets the fragment will cause an error
+  if (assets.length === 0) {
```

Comment:
I don't believe this check is sufficient, as existingNodes aren't checked.  The mock asset would end up available to query in real data sets when the sync detects no changes.